breadth_first_traversal/BreadthFirstTraversal.java

```diff
@@ -0,0 +1,64 @@
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Set;
+
+public class BreadthFirstSearch {
+
+    // Array  of lists for Adjacency List Representation
+    public static LinkedList<Integer>[] adj;
```

Comment:
@jsroyal For templates, refer to https://github.com/iiitv/algos/blob/master/linked_list/LinkedList.java for an example.